core/src/main/scala/org/scalafmt/util/TreeOps.scala

```diff
@@ -229,6 +229,25 @@ object TreeOps {
     case _ => None
   }
 
+  /**
+   * Whether the [[tree]] is a class, trait or def
+   *
+   * For classes this includes primary and secondary Ctors.
+   */
+  def isClassOrDef(tree: Tree): Boolean = tree match {
```

Comment:
Type applications should probably not be included in isDefnSite, I opened https://github.com/scalameta/scalafmt/issues/867 to remind myself to investigate that and fix `isDefnSite`. For now, I think it's OK for you to create your own utility for vertical multiline.

`isClassOrDefn` is not so accurate here since this also includes `Defn.Trait`. A more appropriate name could be `isDefnSiteWithParams`, since it excludes `Defn.Val` which is a definition but has no parameters. It's up to you what name you prefer.